### PR TITLE
Fix bad data in the contact list

### LINKED ISSUE #845: Home feed doesn't show recent notes
Rabble has been seeing very old content at the top of his home feed recently. It seems to only be happening on accounts that follow a lot of users (Rabble follows 700).
Steps to reproduce: 
1. Follow 700 people
2. Launch Nos
Expected: The home feed shows content from today
Actual: The home feed often shows content that is over 24 hours old.

### PR DESCRIPTION
**Pull request by mplorentz**
This should fix #845. A description of the bug is in the comments on the ticket. The fix was to validate that `Author` keys are valid hexadecimal public keys before sending them to relays. I also added validation before writing Follows to the database. 

Writing the validation logic led me down a path of refactoring. I started adding an `isValid` property to `HexadecimalString`, but I realized that we weren't just validating that the string was hexadecimal, but also of the length we expect for a Nostr identifier. So I renamed `HexadecimalString` to `RawNostrID`. Then just for fun I typealiased `RawNostrID` into `RawEventID` and `RawAuthorID`. These types aren't strict (you can instantiate an invalid one) but it does help us communicate in our APIs what type of ID we expect. They also play nicely with Core Data. Maybe someday we should refactor these into a stronger type but I didn't want/need to do it now. 

I also added an analytics event to log "bad req:" errors from relays. This sort of message isn't standard in the protocol but several relays were sending it and that's how I eventually found this bug. This will help us see if there are other bugs like this in the wild.

### CODE REVIEW

**Comment by rabble** [comment]
I tested this with a build on my phone with the bad follow data and for me this PR fixes my problems!

**Review by joshuatbrown — CHANGES_REQUESTED** [review]
I ran the app and it works for me, but I was never able to reproduce the issue.

**Review by martindsq — APPROVED** [review]
It's working fine for me

**Comment by mplorentz** [comment]
@joshuatbrown could you address the remaining feedback here and get this merged? I'm assigning the ticket to you.

File: Nos/Models/CoreData/Follow+CoreDataClass.swift [hunk 1]
```diff
@@ -61,6 +61,7 @@ struct FollowerComparator: SortComparator {
 @objc(Follow)
 public class Follow: NosManagedObject {
     
+    // swiftlint:disable:next function_body_length
```
[the remark this answers is not in the document]
Comment (joshuatbrown) [reply Nos/Models/CoreData/Follow+CoreDataClass.swift:64]: For now, let's just let it slide.

**Review by joshuatbrown — APPROVED** [review]

**Pull request merged**